frontend/src/field_with_permission.tsx

```diff
@@ -24,15 +24,46 @@ const useStyles = makeStyles({
   }
 });
 
+/**
+ * Regexp validation of the supported types, email, tel, url, and text, where
+ * empty values are always valid, and any text value is valid.
+ *
+ * @param fieldType
+ * @param value
+ */
+function validateType(fieldType: LimitedInputType, value: string | null): boolean {
+  if (!value) {
+    return true
+  }
+  switch (fieldType) {
+    case 'email':
+      // Validation with RFC5322
+      return !!value.match(/(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/);
```

Comment:
Why not use a dependency here? Perhaps https://www.npmjs.com/package/isemail

Reply:
I guess we could - but does it provide any functionality that we need, that we don't get in this regexp?

Reply:
The benefit is that you won't need to test.

Reply:
What about that Formik library? or is that another use-case?

Reply:
Formik and yup?

Reply:
yup yup has an email validator, they don't have any tests for it except that the empty string is not an email

Reply:
ah well...

Reply:
Formik email validation seems to be based on giving your own regexp though... and the one in their example will fail for tons of valid emails. Isemail seems to be equivalent to the regexp, but I can change it out since dependencies are so popular! :)